homeassistant/components/electric_kiwi/__init__.py

```diff
@@ -0,0 +1,68 @@
+"""The Electric Kiwi integration."""
+from __future__ import annotations
+
+import aiohttp
+from electrickiwi_api import ElectricKiwiApi
+from electrickiwi_api.exceptions import ApiException
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryAuthFailed, ConfigEntryNotReady
+from homeassistant.helpers import aiohttp_client, config_entry_oauth2_flow
+
+from . import api
+from .const import DOMAIN
+from .coordinator import ElectricKiwiHOPDataCoordinator
+
+PLATFORMS: list[Platform] = [
+    Platform.SENSOR,
+]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Electric Kiwi from a config entry."""
+    implementation = (
+        await config_entry_oauth2_flow.async_get_config_entry_implementation(
+            hass, entry
+        )
+    )
+
+    session = config_entry_oauth2_flow.OAuth2Session(hass, entry, implementation)
+
+    try:
+        await session.async_ensure_token_valid()
+    except aiohttp.ClientResponseError as err:
+        if 400 <= err.status < 500:
+            raise ConfigEntryAuthFailed(err) from err
+        raise ConfigEntryNotReady from err
+    except aiohttp.ClientError as err:
+        raise ConfigEntryNotReady from err
+
+    ek_api = ElectricKiwiApi(
+        api.AsyncConfigEntryAuth(aiohttp_client.async_get_clientsession(hass), session)
+    )
+    hop_coordinator = ElectricKiwiHOPDataCoordinator(hass, ek_api)
+
+    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = {
+        "hop_coordinator": hop_coordinator,
+    }
```

Comment:
We don't have to create a dictionary if there is a single entry in it.

```suggestion
    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = hop_coordinator
```

Additionally, this must be moved down until after the first refresh has completed. Otherwise we may end up with object in hass data that aren't viable (e.g., if `ConfigEntryNotReady was thrown).

Reply:
FYI next PR will contain an additional coordinator. My HACS version already does. It is a bit of work to change this only to change it back on the next PR. I'll move it, but can I keep the dictionary please?

Reply:
Let's not add things now that aren't needed. If a follow up requires more to be stored, we can do that at that point. In this PR it has no use. Please remove the mapping.